File: conciertosBack/api/src/main/java/com/rubenazo/buscaConciertos/adapters/in/dto/ConcertDto.java
package com.rubenazo.buscaConciertos.adapters.in.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.rubenazo.buscaConciertos.domain.Concert;

import java.util.List;

public record ConcertDto(
    String id,
    @JsonProperty("salaConcierto_id") String salaConciertoId,
    @JsonProperty("artist_ids") List<String> artistIds,
    String date,
    String time,
    String price,
    @JsonProperty("source_url") String sourceUrl,
    @JsonProperty("updated_at") String updatedAt
) {
    public static ConcertDto from(Concert concert) {
        return new ConcertDto(
            concert.id(),
            concert.salaConciertoId(),
            concert.artistIds(),
            concert.date().toString(),
            concert.time(),
            concert.price(),
            concert.sourceUrl(),
            concert.updatedAt().toString()
        );
    }
}

File: conciertosFront/shared/src/commonMain/kotlin/org/rubenazo/conciertosfront/core/data/remote/dto/ConcertDto.kt
package org.rubenazo.conciertosfront.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ConcertDto(
    @SerialName("id") val id: String,
    @SerialName("salaConcierto_id") val salaConciertoId: String,
    @SerialName("artist_ids") val artistIds: List<String>,
    @SerialName("date") val date: String,
    @SerialName("time") val time: String?,
    @SerialName("price") val price: String?,
    @SerialName("source_url") val sourceUrl: String?,
    @SerialName("updated_at") val updatedAt: String
)
